### ISSUE TITLE
Style Guide: Method Invocation is outdated

### ISSUE THREAD

**Issue by Sporarum** [scala/docs.scala-lang#2827]
For example:
"""
Scala allows the omission of parentheses on methods of arity-0 (no arguments):
```scala
reply()
// is the same as
reply
```
"""
This is not allowed, at least in Scala 3:
https://scastie.scala-lang.org/8HKODlg1QXWvKbo0QfeLvw

There is also no mention of the `inline` keyword, which I believe to be now required for inline methods

**Comment by som-snytt** [comment]
The truth is rather subtle due to the Java exclusion and empty application not entirely excluded.

**Comment by Sporarum** [comment]
Yes yes of course, but that doesn't change the fact the page should be updated ^^
